include/bluetooth/direction.h

```diff
@@ -26,6 +26,87 @@ struct bt_df_adv_cte_tx_param {
 	uint8_t  *ant_ids;
 };
 
+enum BT_DF_CTE_RX_TYPE {
+	BT_DF_CTE_RX_TYPE_AOA = BIT(0),
+	BT_DF_CTE_RX_TYPE_AOD_1US = BIT(1),
+	BT_DF_CTE_RX_TYPE_AOD_2US = BIT(2),
+};
```

Comment:
Does this differ from the GAP enum? 
```
/** Constant Tone Extension (CTE) types */
enum {
	/** Angle of Arrival */
	BT_GAP_CTE_AOA = 0x00,
	/** Angle of Departure with 1 us slots */
	BT_GAP_CTE_AOD_1US = 0x01,
	/** Angle of Departure with 2 us slots */
	BT_GAP_CTE_AOD_2US = 0x02,
	/** No extensions */
	BT_GAP_CTE_NONE = 0xFF,
};
```

Reply:
The `enum BT_DF_CTE_RX_TYPE` members values are bit field values. They are used to set `bt_df_per_adv_sync_cte_rx_param::cte_type`. Why bit field, because an application may expect to receive IQ samples report for CTE transmitted in AoA and AoD(1us, 2us) modes. At least BT 5.1 spec allows for that.

Reply:
Enum defined in `gap.h` seems to be not used anywhere now. What more, in my humble opinion, it belongs more to direction finding than to GAP. 
`HCI_LE_Periodic_Advertising_Create_Sync` command, that seems to be related with enum from gap.h, also uses bitfield for `sync_cte_type` parameter. Actually `sync_cte_type` has more allowed values, `BT_DF_CTE_RX_TYPE` is a subset of those. @Thalley correct me if I'm wrong, please.

Reply:
I've given a thought to @joerchan proposal to unify both enums. I suggest is: don't do it. 
Enum provided in gap.h has different purpose and there is missing one value: `BT_GAP_CTE_ONLY`. The purpose of the enum seems to be limiting a periodic advertising syncs depending of CTE type.

The CTE types used by Direction finding RX and TX are subset of the CTE type defined in gap.h. But pay attention that DF does not use something like `CTE_NONE` or `CTE_ONLY` and that may be confusing for app developers.

I will unify CTE types but only within DF implementation. I'll create an issue in github to update DF connectionless TX parameters to use the unified version of CTE types after this PR is merged.

@Thalley @joerchan what do you think about it?

Reply:
The GAP enum is used for the `struct bt_le_per_adv_sync_recv_info` at least, among others (or at least that was the intention). 

I understand why you argue for it to be in direction finding, which would be the only implemented usecase of it (and as far as I know, the only SIG defined use case), but nonetheless it is a generic feature that may be used in the application in other ways. I had originally defined them in bluetooth.h but moved them to gap.h on @joerchan's request. 

Regardless, I believe that it makes the most sense of having a single definition for these CTE types for the public API.  How they are transformed/used towards HCI can be dealt with in other ways. Our goal should be to keep the public API simple.

Reply:
Ok, I'll create a common enum that will be used in this PR and also for `struct bt_le_per_adv_sync_recv_info`. The enum will be a bit-field, because I want to use it also for RX part of DF where more than one type of CTE may be expected. 
That will not prevent use of the enum values as integers for CTE type in DF TX and periodic advertising reports. 
Update of DF TX and periodic advertising will be done in separate PR. Then I'll also remove the enum from gap.h.